src/Discord.Net.WebSocket/DiscordSocketApiClient.cs

```diff
@@ -191,24 +211,41 @@ internal override async Task ConnectInternalAsync()
                 if (WebSocketClient != null)
                     WebSocketClient.SetCancelToken(_connectCancelToken.Token);
 
-                if (!_isExplicitUrl)
+                string gatewayUrl;
+                if (_resumeGatewayUrl == null)
+                {
+                    if (!_isExplicitUrl)
+                    {
+                        // TODO: 'GetGatewayAsync' -> 'GetBotGatewayAsync', but it could just be hardcoded to 'wss://gateway.discord.gg/'
+                        var gatewayResponse = await GetGatewayAsync().ConfigureAwait(false);
```

Comment:
```suggestion
                        var gatewayResponse = await GetBotGatewayAsync().ConfigureAwait(false);
```

From DV8:
> - new sessions go to /bot/gateway
> - resumable session to to resume_url
> - if a resume fails, you need a new session

Reply:
It's the same thing, i.e. `wss://gateway.discord.gg/`.
I'd just hardcode it, which would also solve the issue of individual shards all re-requesting this value redundantly.

Reply:
Do you wanna code this in?

Reply:
You mean hardcode it?